### PR TITLE
[4.12] OCPBUGS-6789: Enforce PSA when techpreview is enabled

### PR DESCRIPTION
**Pull request by stlaz**
backport of https://github.com/openshift/cluster-kube-controller-manager-operator/pull/663

/cc @deads2k @ibihim

### CODE REVIEW

**Comment by deads2k** [comment]
/approve
/assign @ibihim 

for lgtm

File: pkg/operator/targetconfigcontroller/targetconfigcontroller.go [hunk 1]
```diff
@@ -172,6 +172,10 @@ func isRequiredConfigPresent(config []byte) error {
 
 	requiredPaths := [][]string{
 		{"extendedArguments", "cluster-name"},
+
+		// featuregates are needed to ensure proper security is enabled.
+		{"extendedArguments", "feature-gates"},
+		{"featureGates"},
```
Comment (ibihim) [inline pkg/operator/targetconfigcontroller/targetconfigcontroller.go:178]: Two different variants exists due to https://github.com/openshift/cluster-kube-controller-manager-operator/pull/695. But they are done at the same time?

Comment (stlaz) [reply pkg/operator/targetconfigcontroller/targetconfigcontroller.go:178]: they each belong to a different component, see the comments in `pkg/operator/configobservation/configobservercontroller/observe_config_controller.go`

**Review by ibihim — APPROVED** [review]

**Comment by stlaz** [comment]
/retest-required
unrelated

**Comment by ingvagabund** [comment]
@stlaz is this PR still relevant?

**Pull request merged**